tests/testdocs.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+# Go through the documentation build process without tampering with any existing files
+
+UUID=$(cat /proc/sys/kernel/random/uuid)
+TDOCS_BKUP="sphinx-source-modules.runtestsbackup.temp.$UUID"
+TDOCS_MODULES="sphinx/source/modules"
+# Include dotted files in globbing
+shopt -s dotglob
+# Make a backup folder
+mkdir $TDOCS_BKUP
+# Back up everything that is currently in the modules folder
+if [ -e $TDOCS_MODULES ]
+then
+    mv $TDOCS_MODULES $TDOCS_BKUP
+fi
+# Create a fresh folder
+mkdir $TDOCS_MODULES
+# Build the document tree
+sphinx-apidoc itsim -METf -o $TDOCS_MODULES
```

Comment:
So as to fail faster whenever `sphinx-apidoc` fails:
```suggestion
sphinx-apidoc itsim -METf -o $TDOCS_MODULES && make -C sphinx html
```